### ISSUE TITLE
Add Experiment Tracking/Tracer Support (MLflow, TensorBoard, W&B)

### ISSUE THREAD

**Issue by HernandoR** [tracel-ai/burn#3416]
### Feature description

It would be nice for Burn to support experiment tracking through popular tracers like MLflow, TensorBoard, and Weights & Biases. This would include:

- Logging training metrics (loss, accuracy, etc.) remotly
- Tracking hyperparameters
- Saving model artifacts
- Visualizing training progress
- Comparing multiple runs

### Feature motivation

- Research Productivity: Currently requires hand-craft logging or custom solutions,.
- Model Governance: Need proper versioning of experiments in production workflows
- Collaboration: Teams need shared platforms to compare results
- Competitiveness: All major DL frameworks (PyTorch, TF) have built-in tracer support

#### Core Design:
``` 

pub trait Tracer {
    fn log_scalar(&self, name: &str, value: f32, step: usize);
    fn log_histogram(&self, name: &str, values: &[f32], step: usize);
    fn log_params(&self, params: &HashMap<&str, Value>);
    // ... other logging methods
}

// Example backends
pub struct TensorBoardTracer { /* ... */ }
pub struct MlflowTracer { /* ... */ }
pub struct WandbTracer { /* ... */ }
``` 
Example Usage:
```
let tracer = BurnTracer::new()
    .with_tracer(
        TensorBoardTracer::new("runs/exp1")
    )
    .with_tracer(
        MlflowTracer::new("http://localhost:5000")
            .experiment("mnist")
    );

tracer.log_params(&model_config);
trainer.run(&mut model, tracer);
```

### Alternative:
It may be easier for the Tracers to support us. But based on my experience, the users will have to align the interface in the project. It would be nice if we chould provide a unified interface.

**Comment by laggui** [comment]
Duplicate of #382 

@antimora we can close one of the two issues, maybe the older one. But since you originally opened it I'll let you decide 🙂

**Comment by HernandoR** [comment]
I notice that there are few implementations of mlflow [trs-flow](https://crates.io/crates/trs-mlflow) [mlflow-client](https://crates.io/crates/mlflow-client)  tensorboard[tensorboard-rs](https://crates.io/crates/tensorboard-rs); 

I'd like to clarify that this issue about tracking to existing tracking platforms (just like #382) rather than creating a new one. Which probably a separate Project.
